code/game/objects/structures/stool_bed_chair_nest/bed.dm

```diff
@@ -103,7 +103,15 @@
 				"You unbuckle yourself from \the [src].",
 				"You hear metal clanking.")
 		playsound(src, 'sound/misc/buckle_unclick.ogg', 50, 1)
-		return TRUE
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			if(H.disabilities & ANEMIA)
+				if(world.time - H.lastAnemia <= 30 SECONDS)
+				else
+					H.sleeping += 2
```

Comment:
you can cause a collapse without brief sleeping by using `H.Knockdown(2)` instead

Reply:
The brief sleeping is intentional

Reply:
as exxion said, the sleep is intentional. it's my experience that when i (anemic) stand up too fast, i essentially pass out and my vision goes black. i originally had it on knockdown, but i felt that sleep captured that aspect of it better because it introduces the black screen.

Reply:
It is too bad about the tick delay but there's not currently an easy way around that for sleeping in particular